### PR TITLE
Add support for AWS::IoT::AccountAuditConfiguration

### PR DESCRIPTION
**Pull request by anton-aws**
### Notes

AccountAuditConfiguration is a Device Defender resource for configuring audit settings for the account. API documentation can be found at https://docs.aws.amazon.com/iot/latest/apireference/API_UpdateAccountAuditConfiguration.html. 
This resource is quite special, here some highlights: 
1. There's just 1 configuration per account - so we introduced accountId as the primaryIdentifier.
1. Describe API always returns a map with all checks - untouched ones are disabled.
1. There's no Create API.
1. There is no clientRequestToken and there are no tags.

### Testing
* Unit tests - 80%+ coverage.
* Contract tests passed - except the false negatives that we're aware of (because of AccountAuditConfiguration intricacies).
* Ran `pre-commit run --all-files`, `mvn verify` successfully.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        }
+      }
+    },
+    "AuditNotificationTarget": {
+      "type": "object",
+      "properties": {
+        "TargetArn": {
+          "description": "The ARN of the target (SNS topic) to which audit notifications are sent.",
+          "type": "string",
+          "maxLength": 2048
```
Comment (vabhishe93) [inline aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json:22]: Minor: Would it be better to keep this consistent with our CP definition:
https://code.amazon.com/packages/IotSepioServiceModel/blobs/df8766043d97ea106eaa742aa63f09c6ec6b5779/--/model/common-types.xml#L186

File: aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        },
+        "Enabled": {
+          "description": "True if notifications to the target are enabled.",
+          "type": "boolean"
+        }
+      }
+    }
+  },
+  "properties": {
+    "AccountId": {
```
Comment (vabhishe93) [inline aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json:38]: Where do we specify that this field would determine whether this resource in unique in CFN universe or not?

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        // Note that this API never throws ResourceNotFoundException.
+        // If the configuration doesn't exist, it returns an object with all checks disabled,
+        // other fields equal to null.
+        DescribeAccountAuditConfigurationResponse describeResponse;
+        try {
+            describeResponse = proxy.injectCredentialsAndInvokeV2(
+                    DescribeAccountAuditConfigurationRequest.builder().build(),
+                    iotClient::describeAccountAuditConfiguration);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
```
Comment (vabhishe93) [inline aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java:58]: Will these exceptions be logged somewhere for debugging purposes

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+            // We return success if the existing configuration is identical.
+            // We do this to not break the customer's template in case of a transient exception.
+            // Consider a scenario where our CreateHandler succeeds, but our response doesn't reach CFN.
+            // CFN would retry. If we returned AlreadyExists, CFN would throw "CREATE_FAILED: resource already exists"
+            // to the customer.
+            // We do know that this configuration could've been created manually rather than through
+            // CFN, but, unlike for other resources, we don't have a way to tell how
+            // the configuration was created.
+            boolean areEquivalent = areEquivalent(model, describeResponse, logger);
+            if (areEquivalent) {
```
Comment (vabhishe93) [inline aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java:78]: Maybe we should add some logs to show which path we entered.

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/DeleteHandler.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        logger.log("Called DescribeAccountAuditConfiguration for " + accountId);
+
+        // We judge whether the configuration exists by the RoleArn field.
+        // For an existing configuration, the RoleArn can never be nullified,
+        // unless the whole configuration is deleted.
+        if (StringUtils.isEmpty(describeResponse.roleArn())) {
+            return ProgressEvent.failed(request.getDesiredResourceState(), callbackContext,
+                    HandlerErrorCode.NotFound,
+                    "The configuration for your account has not been set up or was deleted.");
+        }
```
Comment (vabhishe93) [inline aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/DeleteHandler.java:54]: What happens if CFN's call to lambda fails and then next retry attempt hits this case? Would CFN consider this request to have succeeded considering error code is "NotFound"

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/UpdateHandler.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        // The UpdateAccountAuditConfiguration API has a PATCH behavior, so we can't just
+        // translate the map from the model.
+        // A customer can enable a check out of band that this template doesn't mention.
+        // We create the map in 2 steps. First, create a map with all supported checks disabled.
+        // Second, copy all the entries from the model.
+        Map<String, AuditCheckConfiguration> checkConfigurationsForUpdate = new HashMap<>();
+        describeResponse.auditCheckConfigurations().keySet().forEach(k ->
+                checkConfigurationsForUpdate.put(k, AuditCheckConfiguration.builder().enabled(false).build()));
+        model.getAuditCheckConfigurations().forEach((key, value) ->
+                checkConfigurationsForUpdate.put(key, Translator.translateCheckConfigurationFromCfnToIot(value)));
```
Comment (vabhishe93) [inline aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/UpdateHandler.java:98]: This means that any out of band changes will be overriden. Does this match cloudformation's guidance for out of band changes?

File: aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json (hunk 2 above)
Comment (anton-aws) [reply aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json:38]: Below, there's:
```
"primaryIdentifier": [
  "/properties/AccountId" 
]
```

Verified with a manual test that CFN ensures the config is unique in its universe. First created one stack, then tried to create another one and CFN threw "<accountId> already exists in stack <stack1-arn>"

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java (hunk 3 above)
Comment (anton-aws) [reply aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java:58]: Yes, the framework logs the exception and the stack trace.

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/DeleteHandler.java (hunk 5 above)
Comment (anton-aws) [reply aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/DeleteHandler.java:54]: CFN delete must succeed despite this exception - saw a confirmation on this from a CFN engineer + there's a documented contract `A delete handler MUST return FAILED with a NotFound error code if the resource did not exist prior to the delete request`.

Also did a manual test where I deleted the config out of band, then deleted via CFN and it still succeeded.

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/UpdateHandler.java (hunk 6 above)
Comment (anton-aws) [reply aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/UpdateHandler.java:98]: (Discussed offline and agreed) CFN doesn't provide guidance on this, they leave this up to services. We'll stay consistent across the Device Defender resources and will treat the template as the authoritative desired configuration, overriding changes made out-of-band.

File: aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java (hunk 4 above)
Comment (anton-aws) [reply aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java:78]: Sounds good, added new logs.

**Review by vabhishe93 — APPROVED** [review]

File: aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json (hunk 2 above)
Comment (svarvadekar) [reply aws-iot-accountauditconfiguration/aws-iot-accountauditconfiguration.json:40]: nice

**Review by svarvadekar — APPROVED** [review]

**Review by saurabh14rajput — APPROVED** [review]

**Pull request merged**